content/shmem_broadcast.tex

```diff
@@ -72,12 +72,15 @@
     needed to ensure this): The \VAR{pSync} array on all \ac{PE}s in the
     \activeset{} is not still in use from a prior call to a broadcast routine.  The
     \dest{} array on all \ac{PE}s in the \activeset{} is ready to accept the
-    broadcast data.
+    broadcast data. \newtext{The data in the \source{} array on the root \ac{PE} is ready to be broadcasted.}
+    
+    \newtext{The \source{} and \dest{} arrays may be the same array, but they may not be overlapping arrays. If they 
+    are different, the source array on the non-root \acp{PE} and \dest{} array on the root \ac{PE} is not used by the broadcast routine.}
```

Comment:
> if they are different, the source array on the non-root \acp{PE} and \dest{} array on the root \ac{PE} is not used by the broadcast routine.

This particular statement would cause performance impact - esp on implementations like Cray SHMEM which uses network assisted multicast operations for performing SHMEM BCAST operations. SHMEM has a unique scenario where we can't BCAST to the root destination array. Most HW BCAST operations can't handle this (let others confirm as well). Hence implementations have some sort of temporary array with the value before the HW BCAST operations and then copy back the data to the destination array just in the root PE. We can't use the pSync for this purpose, because we would need pSync to store the temporary state.    

If we need this restriction specifically, we should try to change/remove the existing statement: "The data is not copied to the dest area on the root PE."